widget_bandsplot/bandsplot.py

```diff
@@ -24,6 +28,18 @@ def hex_alpha_to_rgba(color_hex: str, alpha: float = None) -> str:
     return f"rgba{str(rgba)}"
 
 
+def generate_random_colors(num: int) -> list:
+    colors = []
+
+    for _ in range(num):
+        x = "#" + "".join([random.choice("ABCDEF0123456789") for i in range(6)])
+        y = "#" + "".join([random.choice("ABCDEF0123456789") for i in range(6)])
```

Comment:
If I understand it correctly, the x,y are for up/down spins? Since they are already distinguished by line type, I suggest keeping the color the same.

Reply:
And I think it is more clear to generate random color by:

```python
import random
color = "%06x" % random.randint(0, 0xFFFFFF)
```

Reply:
See the comment, I think rather than using the random color we just provide two default color.

Reply:
As you suggested, I provided four default colors for the bandstructure data, which are "black“，"red", "blue", "yellow".